### PR TITLE
Additional plot line styling options

### PR DESCRIPTION
**Pull request by kaperoo**
## Description
This PR adds additional line styling options.

As requested in [DSEGOG-192](https://stfc.atlassian.net/browse/DSEGOG-192), the user can now change the marker style (and size):

Currently the marker size range is set to 1-10.
Available marker styles: `circle`, `cross`, `crossRot`, `dash`, `line`, `rect`, `rectRounded`, `rectRot`, `star`, `triangle` and `None`.

User can also change the line width:

Again, the range is set to 1-10.

~~Additionally, marker colours can be changed independently of the plot line:~~

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
connect to DSEGOG-192

[DSEGOG-192]: https://stfc.atlassian.net/browse/DSEGOG-192?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/plotting/plotSettings/moreOptions/numberInput.component.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import {
+  Unstable_NumberInput as BaseNumberInput,
+  NumberInputProps,
+} from '@mui/base/Unstable_NumberInput';
+import { styled } from '@mui/system';
+import ArrowDropUpIcon from '@mui/icons-material/ArrowDropUp';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+
+const NumberInput = React.forwardRef(function CustomNumberInput(
```
Comment (louise-davies) [inline src/plotting/plotSettings/moreOptions/numberInput.component.tsx]: Why are we using this custom `NumberInput`? I feel `Textfield` with `type="number"` is sufficient for us, like we use in the shotnumber picker.

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx [hunk 2]
```diff
@@ -176,53 +374,74 @@ const MoreOptionsBox = (props: MoreOptionsProps) => {
...
+            display: 'flex',
+            flexDirection: 'row',
+            width: 'inherit',
+            justifyContent: 'space-between',
+            border: 1,
+            padding: 1,
+            alignItems: 'center',
+          }}
+        >
+          <Typography sx={{ fontSize: 12 }}>Marker colour</Typography>
```
Comment (louise-davies) [inline src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx]: Not sure I'm a fan of the marker colour option - you rarely see scientific plots with marker colours different to the line colour, so this is just taking up space in the config box I feel

**Review by louise-davies — CHANGES_REQUESTED** [review]
Also, I feel that with the more options, we should change `moreOptionsToggle` to have `bottom: -20` instead of `top: -95` - i.e. it should pop up just a little lower than the menu, as this reduces the amount of scrolling. As an aside, I think we hardcode a white background in `moreOptionsToggle`, so changing `backgroundColor: '#fff'` to `backgroundColor: 'background.paper'` would also be helpful!

File: src/plotting/plotSettings/moreOptions/numberInput.component.tsx (hunk 1 above)
Comment (kaperoo) [reply src/plotting/plotSettings/moreOptions/numberInput.component.tsx]: You're right. I don't know why I created a custom one 😅

**Comment by kaperoo** [comment]
Changed the custom `NumberInput` to MUI's TextField, removed the _marker colour_ option and made the requested formatting changes.

File: src/plotting/plotSettings/moreOptions/colourPicker.component.tsx [hunk 3]
```diff
@@ -7,6 +7,8 @@ type ColourPickerProps = {
   channelName: string;
   colour: string;
   changeColour: (colour: string) => void;
+  marker?: boolean;
+  sameAsLine?: boolean;
```
Comment (louise-davies) [inline src/plotting/plotSettings/moreOptions/colourPicker.component.tsx]: Think this was missed in the refactor - remove

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx [hunk 4]
```diff
@@ -115,7 +91,10 @@ const MoreOptionsBox = (props: MoreOptionsProps) => {
               thisChannel.displayName ?? thisChannel.name
             } visibility ${thisChannel.options.visible ? 'off' : 'on'}`}
             sx={{ m: 1 }}
-            onChange={() => toggleChannelVisibility()}
+            // onChange={() => toggleChannelVisibility()}
```
Comment (louise-davies) [inline src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx]: Remove commented out code

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx [hunk 5]
```diff
@@ -31,27 +36,30 @@ const MoreOptionsBox = (props: MoreOptionsProps) => {
...
-    changeSelectedPlotChannels(newSelectedPlotChannelsArray);
-  }, [changeSelectedPlotChannels, thisChannel.name, selectedPlotChannels]);
-
-  const changeChannelColour = React.useCallback(
-    (selectedColour: string) => {
+  const changeChannelOption = React.useCallback(
+    <K extends keyof SelectedPlotChannel['options']>(
+      option: K,
+      newValue: SelectedPlotChannel['options'][K]
+    ) => {
```
Comment (louise-davies) [inline src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx:57]: Great refactor, good use of typescript type stuff here

File: src/plotting/plotSettings/moreOptions/moreOptionsToggle.component.tsx [hunk 6]
```diff
@@ -35,7 +35,7 @@ const MoreOptionsToggle = (props: MoreOptionsProps) => {
             top: -95,
```
Comment (louise-davies) [inline src/plotting/plotSettings/moreOptions/moreOptionsToggle.component.tsx]: Can we change `top: -95` to `bottom: -18`? This makes it so the popup menu appears mostly above the channel we're editing, which stops any overflow issues

**Review by louise-davies — CHANGES_REQUESTED** [review]
Works well and the code is looking good, just a few minor code comments & a styling comment

**Review by louise-davies — APPROVED** [review]
LGTM :)

**Pull request merged**